templates/rhba70-full-mysql.yaml

```diff
@@ -0,0 +1,693 @@
+    annotations:
+      description: Route for Business Central's https service.
+  spec:
+    host: "${BUSINESS_CENTRAL_HOSTNAME_HTTPS}"
+    to:
+      name: secure-${APPLICATION_NAME}-rhbacentr
+    tls:
+      termination: passthrough
+- kind: Route
+  apiVersion: v1
+  id: "${APPLICATION_NAME}-kieserver-http"
+  metadata:
+    name: "${APPLICATION_NAME}-kieserver"
+    labels:
+      application: "${APPLICATION_NAME}"
+    annotations:
+      description: Route for KIE server's http service.
+  spec:
+    host: "${EXECUTION_SERVER_HOSTNAME_HTTP}"
+    to:
+      name: "${APPLICATION_NAME}-kieserver"
+- kind: Route
+  apiVersion: v1
+  id: "${APPLICATION_NAME}-kieserver-https"
+  metadata:
+    name: secure-${APPLICATION_NAME}-kieserver
+    labels:
+      application: "${APPLICATION_NAME}"
+    annotations:
+      description: Route for KIE server's https service.
+  spec:
+    host: "${EXECUTION_SERVER_HOSTNAME_HTTPS}"
+    to:
+      name: secure-${APPLICATION_NAME}-kieserver
+    tls:
+      termination: passthrough
+- kind: DeploymentConfig
+  apiVersion: v1
+  metadata:
+    name: "${APPLICATION_NAME}-rhbacentr"
+    labels:
+      application: "${APPLICATION_NAME}"
+  spec:
+    strategy:
+      type: Recreate
+    triggers:
+    - type: ImageChange
+      imageChangeParams:
+        automatic: true
+        containerNames:
+        - "${APPLICATION_NAME}-rhbacentr"
+        from:
+          kind: ImageStreamTag
+          namespace: "${IMAGE_STREAM_NAMESPACE}"
+          name: rhba70-businesscentral-openshift:1.0
+    - type: ConfigChange
+    replicas: 1
+    selector:
+      deploymentConfig: "${APPLICATION_NAME}-rhbacentr"
+    template:
+      metadata:
+        name: "${APPLICATION_NAME}-rhbacentr"
+        labels:
+          deploymentConfig: "${APPLICATION_NAME}-rhbacentr"
+          application: "${APPLICATION_NAME}"
+      spec:
+        serviceAccountName: businesscentral-service-account
+        terminationGracePeriodSeconds: 60
+        containers:
+        - name: "${APPLICATION_NAME}-rhbacentr"
+          image: rhba70-businesscentral-openshift
+          imagePullPolicy: Always
+          volumeMounts:
+          - name: businesscentral-keystore-volume
+            mountPath: "/etc/businesscentral-secret-volume"
+            readOnly: true
+          livenessProbe:
+            exec:
+              command:
+              - "/bin/bash"
+              - "-c"
+              - "/opt/eap/bin/livenessProbe.sh"
+          readinessProbe:
+            exec:
+              command:
+              - "/bin/bash"
+              - "-c"
+              - "/opt/eap/bin/readinessProbe.sh"
+          ports:
+          - name: jolokia
+            containerPort: 8778
+            protocol: TCP
+          - name: http
+            containerPort: 8080
+            protocol: TCP
+          - name: https
+            containerPort: 8443
+            protocol: TCP
+          env:
+          - name: KIE_ADMIN_PWD
```

Comment:
The KIE_ADMIN_PWD and KIE_ADMIN_USER should use the same values as the MAVEN_REPO_PASSWORD and MAVEN_REPO_USERNAME respectively or else they will have different generated values

Reply:
Fixed in rhba70-full-mysql.yaml and rhba70-full-mysql-persistent.yaml
          - name: MAVEN_REPO_USERNAME
            value: "${KIE_ADMIN_USER}"
          - name: MAVEN_REPO_PASSWORD
            value: "${KIE_ADMIN_PWD}"

Reply:
This configuration is not valid for case, when is connected an external maven repository. For external maven repository can be set different user and password.

Reply:
@jakubschwan then the maven admin shouldn't have a default and the password not be generated. The original approach prevents the template from working. The description already provides enough information for the user.
@calvinzhuca given @jakubschwan's point you could leave the old values and remove the default value for the admin and password generation, leaving up to the template user to decide which values to provide

Reply:
ok, updated based on you two's feedback